pkg/ccl/backupccl/key_rewriter.go

```diff
@@ -229,6 +229,17 @@ func (kr *KeyRewriter) RewriteKey(key []byte, wallTime int64) ([]byte, bool, err
 	// in which case we are restoring as a system tenant.
 	if kr.fromSystemTenant && bytes.HasPrefix(key, keys.TenantPrefix) {
 		k, ok := kr.tenants.rewriteKey(key)
+		if ok {
+			suffix, _, err := keys.DecodeTenantPrefix(key)
+			if err != nil {
+				return nil, false, err
+			}
+			_, tableID, _ := keys.SystemSQLCodec.DecodeTablePrefix(suffix)
```

Comment:
Even though you're using the system codec to decode a tenant key, this still works because you've stripped the tenant prefix. It also looks like you swallow the error here which implies this is best effort?

Mind adding a comment here that motivates all of this?

Reply:
"We attempt to elide tenant keys from a couple system tables to prevent poor behavior in the restored tenant. This is best effort because the restored tenant can recover even if these keys are restored. Note that we can grab the tenant table ID using the system codec because the tenant prefix has already been stripped from the key."

Reply:
I'm not sure I'd call it 'best effort' -- any key that errors also isn't in one of those tables

Reply:
Added a comment though explaining why we're skipping these

Reply:
Thanks!